src/main/java/com/google/step/finscholar/data/Scholarship.java

```diff
@@ -0,0 +1,310 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.finscholar.data;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+/** The Scholarship Class. */
+public class Scholarship {
+
+  private final String scholarshipName;
```

Comment:
With Builder pattern, all instance variables are declared final, assuming that the object is immutable once declared.